app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionUpdateProfiles.scala

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsstubs.controllers
+
+import java.time.LocalDateTime
+
+import cats.implicits._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import play.api.mvc.Results._
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.BusinessPartnerRecordController.DesErrorResponse
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.SubscriptionController.SubscriptionUpdateResponse
+
+case class SubscriptionUpdateProfile(
+  predicate: String => Boolean,
+  subscriptionUpdateResponse: Either[Result, SubscriptionUpdateResponse]
+)
+
+object SubscriptionUpdateProfiles {
+
+  def updateSubscriptionDetails(id: String): Option[SubscriptionUpdateProfile] =
+    updates.find(_.predicate(id))
+
+  val desUpdate = SubscriptionUpdateResponse(
+    "CGT",
+    LocalDateTime.of(2019, 10, 23, 11, 34, 23).toString,
+    "01234567891",
+    "XACGTP123456701",
+    "GB",
+    Some("TF34NT")
+  )
+
+  private val updates = List(
+    SubscriptionUpdateProfile(
+      _ === "XACGTP123456701",
+      Right(desUpdate)
+    ),
+    SubscriptionUpdateProfile(
+      _ === "XACGTP123456702",
```

Comment:
aren't these the same CGT ref's for the display profiles? I'd have them as different ones otherwise we'd never be able to get to the update part in the FE journey since we won't be able to do display.